acme/acme/messages.py

```diff
@@ -328,12 +331,22 @@ class Registration(ResourceBody):
     @classmethod
     def from_data(cls, phone=None, email=None, external_account_binding=None, **kwargs):
         """Create registration resource from contact details."""
+        # extra variable to detect non-null empty lists or tuples
+        details_provided = 'contact' in kwargs and kwargs['contact'] is not None
+
         details = list(kwargs.pop('contact', ()))
         if phone is not None:
             details.append(cls.phone_prefix + phone)
         if email is not None:
             details.extend([cls.email_prefix + mail for mail in email.split(',')])
-        kwargs['contact'] = tuple(details)
+
+        # Serialization behavior change based on providing `contacts` to the constructor
+        # So we add the `contact` argument if:
+        # - `phone` or `email` or `contact` `kwargs` are passed with values
+        # - a empty list or tuple is provided in the `kwargs`
```

Comment:
I think we should remove or clarify this line. An empty list or tuple is provided in `kwargs` for what?

Reply:
So if the user passes `contact=[]`, we need to replace the `list` with a `tuple` (`list` is not hashable and will throw an exception). This was previously handled by the old code path because the [setting of the `contact` field of kwargs was unconditional](https://github.com/certbot/certbot/blob/master/acme/acme/messages.py#L336) (because explicitly providing an argument didn't change behavior).

However, if the user passes `contact=None`, we need to not set `contact,` because (I didn't actually test this so maybe it's ok?) contact=None will probably save a `null` in the `json` (creating a potential de-serialization problem).